### DISCUSSION TITLE
Sharing results using image generation app!

### DISCUSSION THREAD

**Discussion by ClashSAN** [google-ai-edge/mediapipe-samples#213] [Show and tell]
Thanks to the development team at google, we can test the image generator app for basic txt2img!

The minimum RAM needed on the phone is 6gb, with this you can generate up to `512x512` with some models!
And at 12gb, up to `832x832`!

## `768x768` 20step examples:

| ![Img_2023_08_16_05_13_59](https://github.com/googlesamples/mediapipe/assets/98228077/08973fec-4b9b-4493-923d-89e4040514cb) | ![Img_2023_08_16_20_52_36](https://github.com/googlesamples/mediapipe/assets/98228077/53476ee9-c40c-4f35-b8aa-6f30ef512d2c) |
| ------- | --- |
| ![Img_2023_08_16_04_27_05](https://github.com/googlesamples/mediapipe/assets/98228077/c36931b2-b509-4caa-a078-56afb1d42160) | ![Img_2023_08_16_20_57_43](https://github.com/googlesamples/mediapipe/assets/98228077/14438176-72fb-4f62-8d16-d5ea3df0b8dc) |

## Needs to grab this image with the recents menu screen:

![Screenshot_20230815-225915](https://github.com/googlesamples/mediapipe/assets/98228077/549cce60-5f4d-4d5a-ba9d-57c8d832b32e)

## Users can still make nice images at `384x384`, depending on the model used, and how well the model is trained:

| ![Screenshot_20230816-055438](https://github.com/googlesamples/mediapipe/assets/98228077/a42d812d-9102-4c6e-aa52-0ec6fc05ad50) | ![Screenshot_20230816-073307](https://github.com/googlesamples/mediapipe/assets/98228077/2ee23d06-737f-46dc-86e7-4266fdf02b5d) |
| --- | --- |
| ![Screenshot_20230816-211946](https://github.com/googlesamples/mediapipe/assets/98228077/06a75dad-1652-488c-a365-57fcc72a4e3c) |  |

## Minimum step-count

Here's two different prompts using 8 steps, 14steps, and 20steps. The non-cherrypicked seeds are 20, 22, 24. The first prompt is at `512x512` and the second at `384x384`. Each image was generated on a Pixel 6 Pro,

![2mb](https://github.com/googlesamples/mediapipe/assets/98228077/480bec0e-6ab7-4949-aa0b-850257aaa5e9)

- It remains to be seen how many steps are needed for mediapipe's SOTA 6M parameter control plugins, it may be similar to controlnet in that it requires __fewer__ steps than with the pure txt2img process.

- Improved samplers such as UniPC or DPM++ 2M Karras may further lower the minimum healthy step count to 8-10steps.

- [Distilled Models](https://github.com/segmind/distill-sd) may further lower necessary step count, but come with limitations of needing to be initially pretrained and optimized for certain genre: realism/anime, limiting user-made model options, initial results don't seem to work well with dynamic shapes.

**Comment by sl33pyC01E** [comment]
Can you share a built apk for this?
